www/src/components/docs-table-of-contents.js

```diff
@@ -14,54 +40,71 @@ function isUnderDepthLimit(depth, maxDepth) {
     items &&
-    items.map((item, index) => (
-      <li
-        sx={{ [mediaQueries.xl]: { fontSize: 1 } }}
-        key={location.pathname + (item.url || depth + `-` + index)}
-      >
-        {item.url && (
-          <Link
-            sx={{
-              "&&": {
-                color: `textMuted`,
-                border: 0,
-                transition: t =>
-                  `all ${t.transition.speed.fast} ${t.transition.curve.default}`,
-                ":hover": {
-                  color: `link.color`,
-                  borderBottom: t => `1px solid ${t.colors.link.hoverBorder}`,
+    items.map((item, index) => {
+      const isActive = isDesktop && item.url === `#${activeHash}`
+      return (
+        <li
+          sx={{ [mediaQueries.xl]: { fontSize: 1 } }}
+          key={location.pathname + (item.url || depth + `-` + index)}
+        >
+          {item.url && (
+            <Link
+              sx={{
+                "&&": {
+                  color: isActive ? `link.color` : `textMuted`,
+                  border: 0,
+                  borderBottom: t =>
+                    isActive
+                      ? `1px solid ${t.colors.link.hoverBorder}`
+                      : `none`,
+                  transition: t =>
+                    `all ${t.transition.speed.fast} ${t.transition.curve.default}`,
+                  ":hover": {
+                    color: `link.color`,
+                    borderBottom: t => `1px solid ${t.colors.link.hoverBorder}`,
+                  },
                 },
-              },
-            }}
-            getProps={({ href, location }) =>
-              location && location.href && location.href.includes(href)
-                ? {
-                    style: {
-                      color: colors.link.color,
-                      borderBottom: `1px solid ${colors.link.hoverBorder}`,
-                    },
-                  }
-                : null
-            }
-            to={location.pathname + item.url}
-          >
-            {item.title}
-          </Link>
-        )}
-        {item.items && isUnderDepthLimit(depth, maxDepth) && (
-          <ul sx={{ color: `textMuted`, listStyle: `none`, ml: 5 }}>
-            {createItems(item.items, location, depth + 1, maxDepth)}
-          </ul>
-        )}
-      </li>
-    ))
+              }}
+              to={location.pathname + item.url}
+            >
+              {item.title}
+            </Link>
+          )}
+          {item.items && isUnderDepthLimit(depth, maxDepth) && (
+            <ul sx={{ color: `textMuted`, listStyle: `none`, ml: 5 }}>
+              {createItems(
+                item.items,
+                location,
+                depth + 1,
+                maxDepth,
+                activeHash,
+                isDesktop
+              )}
+            </ul>
+          )}
+        </li>
+      )
+    })
   )
 }
 
-function TableOfContents({ page, location }) {
+function TableOfContents({ page, toc, location }) {
```

Comment:
nits:

* You can get rid of the `page` prop here and replace it with `toc`, and a new `depth` property which takes in `page.tableOfContents.depth`.
* Rename the `toc` prop here to `items` for clarity.

Reply:
@tesseralis Yes, that makes sense. Done in a7dac40 !